### PR TITLE
Suppress MIR comments of FnDef and unit types

### PR DESCRIPTION
**Pull request by tesuji**
An expansion of #75566.
Comments of FnDef MIR constant already contain `ty::Contains` comments.

### CODE REVIEW

**Comment by oli-obk** [comment]
@bors r+

Nice! If you want to do something else along this line, we should really suppress the `const` infront of `FnDef`s

**Comment by lzutao** [comment]
@bors r- 
Build failed. Will fix in tommorow morning.

**Review by tesuji — COMMENTED** [review]
Some tests output are changes. I don't understand why's that?

**Edit**: Everything is normal. I just read the diff wrong.

**Comment by oli-obk** [comment]
r=me with CI passing

**Comment by lzutao** [comment]
CI  is green.

**Comment by tmandry** [comment]
Failed in https://github.com/rust-lang/rust/pull/75712#issuecomment-676580124, @bors r-

**Comment by lzutao** [comment]
`mir-opt/instrument_coverage.rs` test needs profiler support:
https://github.com/rust-lang/rust/blob/5f6fcad6449aabad3c29c65c1738f8eab89c03ce/src/test/mir-opt/instrument_coverage.rs#L5
However, profiler is disable by default:
https://github.com/rust-lang/rust/blob/5f6fcad6449aabad3c29c65c1738f8eab89c03ce/config.toml.example#L220

So the test cannot be run in my machine.

Does profiler need support from LLVM ? I am using local LLVM if that count.

**Comment by lzutao** [comment]
Done!

@rustbot modify labels: -S-waiting-on-author +S-waiting-on-review

File: src/test/mir-opt/array_index_is_temporary.main.SimplifyCfg-elaborate-drops.after.mir.64bit [hunk 1]
```diff
@@ -37,9 +37,6 @@ fn main() -> () {
         StorageLive(_6);                 // scope 4 at $DIR/array-index-is-temporary.rs:16:25: 16:26
         _6 = _3;                         // scope 4 at $DIR/array-index-is-temporary.rs:16:25: 16:26
         _5 = const foo(move _6) -> bb1;  // scope 4 at $DIR/array-index-is-temporary.rs:16:21: 16:27
-                                         // ty::Const
-                                         // + ty: unsafe fn(*mut usize) -> u32 {foo}
```
Comment (bjorn3) [inline src/test/mir-opt/array_index_is_temporary.main.SimplifyCfg-elaborate-drops.after.mir.64bit:41]: The exact type of functions can sometimes be useful.

Comment (oli-obk) [reply src/test/mir-opt/array_index_is_temporary.main.SimplifyCfg-elaborate-drops.after.mir.64bit:41]: The type is still shown right below (though we should probably cut out the `val` part of the `literal` field as it's useless for zsts.

**Pull request merged**